crates/common/api_types/lean/src/block_id.rs

```diff
@@ -0,0 +1,65 @@
+use std::str::FromStr;
+
+use alloy_primitives::{B256, hex};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum BlockID {
+    Finalized,
+    Genesis,
+    Head,
+    Justified,
+    Slot(u64),
+    /// expected to be a 0x-prefixed hex string.
+    Root(B256),
+}
+
+impl Serialize for BlockID {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for BlockID {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        match s.to_lowercase().as_str() {
+            "finalized" => Ok(BlockID::Finalized),
+            "genesis" => Ok(BlockID::Genesis),
+            "head" => Ok(BlockID::Head),
+            "justified" => Ok(BlockID::Justified),
```

Comment:
This code is copy pasted from `pub enum ID {`, instead of copy pasting just make a another api_types crate which is shared between beacon and lean